core/src/main/scala/kafka/server/MetadataCache.scala

```diff
@@ -91,13 +93,10 @@ class MetadataCache(brokerId: Int) extends Logging {
               if (errorUnavailableListeners) Errors.LISTENER_NOT_FOUND else Errors.LEADER_NOT_AVAILABLE
             }
             new MetadataResponse.PartitionMetadata(error, partitionId.toInt, Node.noNode(),
-              Optional.empty(), replicaInfo.asJava, java.util.Collections.emptyList(),
+              Optional.empty(), replicaInfo.asJava, isrInfo.asJava,
```

Comment:
Can we have a test case for this change?

Reply:
Do you think MetadataCacheTest#getTopicMetadataPartitionLeaderNotAvailable cover this?